eng/SnippetGenerator/CSharpProcessor.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace SnippetGenerator
+{
+    public class CSharpProcessor
+    {
+        private static readonly string _snippetFormat = "{3}<code snippet=\"{0}\">{1}{2}</code>";
+        private static readonly Regex _snippetRegex = new Regex("^(?<ident>\\s*)\\/{3}\\s*<code snippet=\\\"(?<name>[\\w:]+)\\\">.*?\\/{3}\\s*<\\/code>",
```

Comment:
Nit - this should be `indent` here and in the match group below.  I was thinking that was some sort of identifier for a minute and pretty confused.